js/understanding-hoisting-the-real-way/understanding-hoisting-the-real-way.md

```diff
@@ -0,0 +1,185 @@
+Hello, I am Shivam Verma. Today we are going to understand one of the most frequently asked questions in interviews. 
+What is **Hoisting** in JavaScript? 
+
+First of all, let us understand the concept of **Hoisting**. Let's Begin
+
+## Understanding "Hoisting"
+
+Let us take a look at a code example.
+
+```javascript
+var x = 7;
+function sayHi() {
+     console.log("Hello World");
+}
+sayHi();
+console.log(x);
+```
+
+Output: 
+
+```javascript
+Hello World
+7
+```
+Simple right.
+ Let's tweak the code a little bit and see
+
+```javascript
+sayHi();
+console.log(x);
+var x = 7;
+function sayHi() {
+     console.log("Hello World");
+}
+```
+
+Output: 
+
+```javascript
+Hello World
+undefined
+```
+
+Well something strange in the output there, isn't it? How is the function executed at the top even before declaration? Why is the value of `x` shown as `undefined`?
+
+This is possible because of **Hoisting**
+
+**Hoisting** is a concept by which you can access variables and functions even before initializing them.
+
+## How does **Hoisting** work?
+
+You might be thinking that the title says "The real way". Hmmm, what is this real way?
+Is **Hoisting** done in more than one way in JavaScript?
+
+No, it's not done in more than one way, there is just some incomplete and little incorrect information spread about **Hoisting**.
+
+Let me clear your doubts. You might have heard the popular definition of **Hoisting** available in numerous blogs and videos on the internet which is somewhat like,
+
+"Variable and function declarations are moved "magically" to the top of the JavaScript code"
+
+Well, we are writing code, not in Hogwarts 😂. Behind every code there lies logic and not magic.
+
+Ok, enough joking, nobody writes magically in the definition, It was put there sarcastically to show why that definition is wrong.
```

Comment:
You can remove the line. The joke was funny but this line is kind of ruining the sarcasm. This is not blogging advice 😅. I'm saying from personal experience.

Sarcasm doesn't need explanation.

Reply:
You can add "Enough with the jokes, let's talk about the real way" and continue with the next para.